app/components/au-address-search.js

```diff
@@ -0,0 +1,383 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { task, timeout } from 'ember-concurrency';
+
+/**
+ * @typedef {Object} Address
+ * @property {string} country
+ * @property {string} province
+ * @property {string} municipality
+ * @property {string} postalCode
+ * @property {string} street
+ * @property {string} houseNumber
+ * @property {string | null} boxNumber
+ * @property {boolean | undefined } completed
+ */
+
+/**
+ * @typedef {Object} LocationInFlanders
+ * @property {string} municipality
+ * @property {string} postalCode
+ * @property {string} street
+ * @property {string} housenumber
+ */
+
+/**
+ * @typedef {'manual'|'automatic'} AddressSearchMode
+ */
+
+/** @type { (fieldName: keyof<Address>) => ((e:UIEvent)=>void) } */
+function handleSimpleManualInputEventChange(fieldName) {
+  return function (e) {
+    e.preventDefault();
+    const newValue = e.target.value ?? undefined;
+    this.selectedAddress = {
+      ...this.selectedAddress,
+      [fieldName]: newValue,
+    };
+  };
+}
+
+export default class AuAddressSearchComponent extends Component {
+  constructor(...args) {
+    super(...args);
+    // We use a task here because by the time ember concurrency invokes it the args will be
+    // passed to the component. When this constructor runs the ember framework has not yet
+    // passed `this.args`
+    this._initTask.perform();
+  }
+
+  /** Initialse the component with args */
+  _initTask = task(async () => {
+    if (this.addressOk) {
```

Comment:
I would initialize all the class properties on top, with a default value to make it more clear